src/sgame/sg_bot_local.h

```diff
@@ -179,6 +179,9 @@ struct botMemory_t
 	int lastNavconTime;
 	int lastNavconDistance;
 
+	void setHasOffmeshGoal( bool val ) { hasOffmeshGoal = val; }
```

Comment:
Do we really need this variable? Why can't we just have action fight give up whenever the target is out of sight AND there is no path?

Reply:
Currently, they will move into the direction of an invisible target even without a path, until `g_bot_chasetime` seconds pass. In the case of mantis, marauder, dragoon this might trigger an upward attack, which can make the bot reach the target. In the other cases, the bot will usually just run against a wall.

This is the current behavior, and I don't want to be responsible for changing it. So I introduced this variable. It is only set for human bots. Maybe it would be better to split the attack action into one function for each team.

Reply:
OK so the current algorithm is to give up the target if it has not been visible for g_bot_chasetime, great. So why can't we just keep the current algorithm for everything and just add the ability to set things without a path as the target?

Reply:
How would that work with aliens which only have melee?
I suspect that if one did the same with aliens as you seem to suggest, we'd soon have dragoon that constantly stuck themselves in places they don't know how to escape from, waiting for a prey to unstuck them.
One would also need a trace for aliens, as their "missiles" (their body) is considerably bigger than any human weapon, so they might be able to see something they don't know how to reach, while being truly unable to attack it still.

The current algorithm is also well known to make bots jump in death pits because they ignore navmesh while fighting their target. If I understand correctly what you suggest, your idea would increase that problem, unlike this PR which makes human bots stand by when they're attacking offmesh targets.

Reply:
> So why can't we just keep the current algorithm for everything and just add the ability to set things without a path as the target?

That would make them move forward when attacking an enemy without a path. But this will often make the bot move to a place where it cannot see the enemy anymore.